lambdas/issue-credential/src/time-handler.ts

```diff
@@ -4,48 +4,56 @@ import { TimeEvent } from "./time-event";
 
 const logger = new Logger();
 
-const Unit = {
-  Seconds: "seconds",
-  Minutes: "minutes",
-  Hours: "hours",
-  Days: "days",
-  Months: "months",
-  Years: "years",
-  None: "none",
-};
+enum Unit {
+  Seconds = "seconds",
+  Minutes = "minutes",
+  Hours = "hours",
+  Days = "days",
+  Months = "months",
+  Years = "years",
+  None = "none",
+}
+
+export class TimeHandler implements LambdaInterface {
+  private readonly timeInSecs: number;
 
-const convert = (unit?: string): number => {
-  switch (unit) {
-    case Unit.Seconds:
-      return 1000;
-    case Unit.Minutes:
-      return 1000 * 60;
-    case Unit.Hours:
-      return 1000 * 60 * 60;
-    case Unit.Days:
-      return 1000 * 60 * 60 * 24;
-    case Unit.Months:
-      return 1000 * 60 * 60 * 24 * 30;
-    case Unit.Years:
-      return 1000 * 60 * 60 * 24 * 365;
-    case Unit.None:
-      return 1;
-    default:
-      throw new Error(`Unexpected time-to-live unit encountered: ${unit}`);
+  constructor() {
+    this.timeInSecs = this.msToSeconds(Date.now());
   }
-};
 
-const parseUnit = (value?: string): string => {
-  const unitKey = Object.keys(Unit).find(
-    (key) => Unit[key as keyof typeof Unit] === value?.toLowerCase()
-  );
-  if (unitKey) {
-    return Unit[unitKey as keyof typeof Unit];
+  private msToSeconds(ms: number): number {
+    return Math.floor(ms / 1000);
+  }
+
+  private convert(unit: Unit): number {
+    switch (unit) {
+      case Unit.Seconds:
+        return 1000;
+      case Unit.Minutes:
+        return 1000 * 60;
+      case Unit.Hours:
+        return 1000 * 60 * 60;
+      case Unit.Days:
+        return 1000 * 60 * 60 * 24;
+      case Unit.Months:
+        return 1000 * 60 * 60 * 24 * 30;
+      case Unit.Years:
+        return 1000 * 60 * 60 * 24 * 365;
+      case Unit.None:
+        return 1;
+      default:
+        throw new Error(`Unexpected time-to-live unit encountered: ${unit}`);
+    }
+  }
+
+  private parseUnit(value?: string): Unit {
```

Comment:
```suggestion
  private parseUnit(value: string): Unit {
```
Looking at the usage of the function, the value will never be undefined